teams-fluid-hello-world/src/client/helloWorldTab/FluidContent.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { Button } from "@fluentui/react-northstar";
+import { SharedMap } from "fluid-framework";
+import { diceValueKey } from "./Util";
+import { useState, useEffect } from "react";
+
+export type FluidContentProps = Readonly<{
+    fluidMap: SharedMap;
+}>
+
+export const FluidContent = (props : FluidContentProps) => {
+
+    const [diceView, setDiceView] = useState<{color : string, content : string} | undefined >(undefined);
+
+    useEffect(() => {
+        // sync Fluid data into view state
+        const updateDice = () => {
+            const diceValue = props.fluidMap.get(diceValueKey);
+            // Unicode 0x2680-0x2685 are the sides of a dice (⚀⚁⚂⚃⚄⚅)
+            setDiceView({
+                content: String.fromCodePoint(0x267f + diceValue),
+                color: `hsl(${diceValue * 60}, 70%, 30%)`
+            });
+        };
+        // ensure update runs at least once
+        updateDice();
+        // Use the changed event to trigger the rerender whenever the value changes.
+        props.fluidMap.on("valueChanged", updateDice);
+
+        // turn off listener when component is unmounted
+        return () => { props.fluidMap.off("valueChanged", updateDice); };
+    }, [props.fluidMap]);
```

Comment:
Since you're passing in the SharedMap here you can get strong typing on this object.

I'm coding this in the browser without validation so please validate.

```suggestion
    const generateCurrentViewState = (): {color : string, content : string} => {
           const diceValue = props.fluidMap.get(diceValueKey);
            // Unicode 0x2680-0x2685 are the sides of a dice (⚀⚁⚂⚃⚄⚅)
            return {
                content: String.fromCodePoint(0x267f + diceValue),
                color: `hsl(${diceValue * 60}, 70%, 30%)`
            };
    }
    const [diceView, setDiceView] = useState<{color : string, content : string}>(generateCurrentViewState());

    useEffect(() => {
        // sync Fluid data into view state
        const updateDice = () => {
            setDiceView(generateCurrentViewState());
        };
        // Use the changed event to trigger the rerender whenever the value changes.
        props.fluidMap.on("valueChanged", updateDice);

        // turn off listener when component is unmounted
        return () => { props.fluidMap.off("valueChanged", updateDice); };
    }, [props.fluidMap]);
```

Reply:
works like a charm!! 💯